packages/web3-eth-contract/src/index.js

```diff
@@ -713,6 +713,10 @@ Contract.prototype.once = function(event, options, callback) {
 Contract.prototype._on = function(){
     var subOptions = this._generateEventOptions.apply(this, arguments);
 
+    if (subOptions.params && subOptions.params.toBlock) {
+        delete subOptions.params.toBlock;
+        console.warn('Invalid option: toBlock. Use getPastEvents for specific range.');
```

Comment:
great!

Reply:
maybe we can add a test to ensure this warning fires?

Reply:
Done